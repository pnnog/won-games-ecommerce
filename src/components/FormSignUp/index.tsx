import TextField from 'components/TextField';
import { Email, Lock } from '@styled-icons/material-outlined';
import Button from 'components/Button';
import Link from 'next/link';
import { FormWrapper, FormLink } from 'components/Form';

const FormSignUp = () => (
  <FormWrapper>
    <form>
      <TextField
        type="email"
        name="email"
        placeholder="Email"
        icon={<Email />}
      />
      <TextField
        type="password"
        name="password"
        placeholder="Password"
        icon={<Lock />}
      />
      <TextField
        type="password"
        name="confirm-password"
        placeholder="Confirm password"
        icon={<Lock />}
      />

      <Button size="large" fullWidth>
        Sign up now
      </Button>

      <FormLink>
        Already have an account? <Link href={'/sign-in'}>Sign in</Link>
      </FormLink>
    </form>
  </FormWrapper>
);

export default FormSignUp;
