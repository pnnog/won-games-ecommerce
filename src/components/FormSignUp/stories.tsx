import { Meta, Story } from '@storybook/react';
import FormSignUp from '.';

export default {
  title: 'Forms/FormSignUp',
  component: FormSignUp,
  args: {},

  parameters: {
    viewport: {
      defaultViewport: 'mobile1',
    },

    layout: {
      default: 'fullscreen',
    },
  },
} as Meta;

export const Default: Story = (args) => (
  <div style={{ width: '30rem', margin: '2rem auto' }}>
    <FormSignUp {...args} />
  </div>
);
