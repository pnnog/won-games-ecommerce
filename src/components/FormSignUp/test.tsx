import { screen } from '@testing-library/react';

import FormSignUp from '.';
import { renderWithTheme } from 'utils/tests/helpers';

//mockar componentes externos e testar
//testar links

jest.mock('components/TextField', () => ({
  __esModule: true,
  default: () => <div data-testid="text field mock"></div>,
}));

jest.mock('components/Button', () => ({
  __esModule: true,
  default: () => <div data-testid="button mock"></div>,
}));

describe('<FormSignUp />', () => {
  it('should render FormSignUp', () => {
    const { container } = renderWithTheme(<FormSignUp />);

    expect(screen.getAllByTestId('text field mock')).toHaveLength(3);
    expect(screen.getByTestId('button mock')).toBeInTheDocument();

    expect(screen.getByRole('link', { name: /Sign in/ })).toHaveAttribute(
      'href',
      '/sign-in'
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
