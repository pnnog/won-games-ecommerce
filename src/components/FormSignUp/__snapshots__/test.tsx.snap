// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormSignUp /> should render FormSignUp 1`] = ` `;
